compiler/rustc_middle/src/mir/query.rs

```diff
@@ -214,6 +214,7 @@ pub struct BorrowCheckResult<'tcx> {
     pub concrete_opaque_types: VecMap<OpaqueTypeKey<'tcx>, Ty<'tcx>>,
     pub closure_requirements: Option<ClosureRegionRequirements<'tcx>>,
     pub used_mut_upvars: SmallVec<[Field; 8]>,
+    pub errored: bool,
```

Comment:
I'm tempted to suggest making the `borrowck` query return `Result<BorrowCheckResult, ErrorReported>` - I don't know if that would regress anything, but if there are any errors I wouldn't want to trust the other fields tbh.